### PR TITLE
Add exercises for Git & Github Workshop Session 2 

### PR DESCRIPTION
**Pull request by bharitwal**
- Rename the branch name in session 1 exercises so that its easier to track based on the sessions.
- Add exercises for session 2

## Type Of Change
- [ ] Update documentation
- [ ] Add new event details
- [X] Add project resources

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Commit message is as per guidelines in the contributor guide

### CODE REVIEW

File: resources/devops/Git & Github/Session2 - Exercises.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Session 2  - Git and Github Exercises
+---
+
+Questions have been designed to be completed in the given sequence and is a continuation of exercises in [Session 1](Session1%20-%20Exercises.md)
+
+1. Change directory to `git-workshop` and investigate the existing branches, commit history and status.
```
Comment (irenkamalova) [inline resources/devops/Git & Github/Session2 - Exercises.md:6]:
```suggestion
1. Change directory to `git-workshop` and investigate the existing branches, the history of commits and status.
```

File: resources/devops/Git & Github/Session2 - Exercises.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <summary>Solution</summary>
+
+    ```bash
+    cd git-workshop
+    git branch
+    git log
+    git status
+    ```
+    </details>
+
```
Comment (irenkamalova) [inline resources/devops/Git & Github/Session2 - Exercises.md:18]: Do we want to provide a solution for this step only?
I see 2 options:
- don't provide a solution at all
- provide solution to every step

Wdy?

File: resources/devops/Git & Github/Session2 - Exercises.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+2. Create a new branch `session2` from the `session1` branch. Switch to the `session2` branch. Investigate where the HEAD is pointing to.
+3. Add a new file `session2-file1` in the `session2` branch. Commit this new file to your local repository.
+4. Edit the file `session2-file1` and add "SESSION2-RESTORE-STAGING" to the file. Add these changes to the staging area (do not commit the changes at this stage). Check the status of your local repository.
+5. Unstage the changes in the file `session2-file1`. Check the status of your local repository.
+6. Now Undo the changes in the file `session2-file1` and restore the file to HEAD. Check the status of your local repository. (Make sure that there are no changes to be committed)
```
Comment (irenkamalova) [inline resources/devops/Git & Github/Session2 - Exercises.md:23]:
```suggestion
6. Undo the changes in the file `session2-file1` and restore the file to HEAD. Check the status of your local repository. Make sure that there are no changes to be committed.
```

File: resources/devops/Git & Github/Session2 - Exercises.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+7. Create 8 new commits in the branch `session2`. These can be any random commits that you can track easily. For simplicity you can use the format that was used during the session 2 event - Enter numerical values (ONE, TWO, THREE, etc) in the file `session2-file1` and commit after adding each value. Use commit messages that are easier to track ( Commit 1, Commit 2, Commit 3, etc). Check the commit history.
+8. Undo the last 2 commits from the local repository such that the changes made are preserved in the working directory. Check git history and status. Undo the changes that exist in the working directory. (You should see 6 new commits in the log at this stage)
```
Comment (irenkamalova) [inline resources/devops/Git & Github/Session2 - Exercises.md:25]:
```suggestion
8. Undo the last 2 commits from the local repository such that the changes made are preserved in the working directory. Check git history and status. Undo the changes that exist in the working directory. You should see 6 new commits in the log at this stage.
```

File: resources/devops/Git & Github/Session2 - Exercises.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+9. Now undo the latest 2 commits without preserving any changes made in these commits to the local directory. (You should only see 4 new commits in the log at this stage)
```
Comment (irenkamalova) [inline resources/devops/Git & Github/Session2 - Exercises.md:26]:
```suggestion
9. Undo the latest 2 commits without preserving any changes made in these commits to the local directory. You should only see 4 new commits in the log at this stage.
```

File: resources/devops/Git & Github/Session2 - Exercises.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+10. Undo the latest 2 commits by preserving the git history that these commits were made and reverted.
```
Comment (irenkamalova) [inline resources/devops/Git & Github/Session2 - Exercises.md:27]: Bonus exercise: `amend` commit and collapse all commits in 1?

**Review by irenkamalova — COMMENTED** [review]
I suggest to move the folder up.
All engineers and even technical writers can benefit from learning git&git hub.
So, let's move it up?

File: resources/devops/Git & Github/Session2 - Exercises.md (hunk 6 above)
Comment (bharitwal) [reply resources/devops/Git & Github/Session2 - Exercises.md:27]: The exercises are based on what was covered during the session. We didn't cover it during this session.

File: resources/devops/Git & Github/Session2 - Exercises.md (hunk 2 above)
Comment (bharitwal) [reply resources/devops/Git & Github/Session2 - Exercises.md:18]: The intention is for volunteers to contribute to these solutions. I only added the first one as a sample format that can be used.

**Comment by bharitwal** [comment]
> 

Most of the topics under devops are generic for all developers (hence the term dev+ops). Future sessions would include -  linux, containers, CI/CD , etc - I think its the depth that one goes into. The intention is to cover all these topics in depth. Hence, I would prefer to keep them under devops.

**Comment by irenkamalova** [comment]
It makes sense to me. It's fair to say that every dev has devops skills 👍  We can keep it under devops folder for now 👌

File: resources/devops/Git & Github/Session2 - Exercises.md (hunk 1 above)
Comment (bharitwal) [reply resources/devops/Git & Github/Session2 - Exercises.md:6]: `commit history` is a pretty standard terminology referenced across all of github documentation : https://docs.github.com/en/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/differences-between-commit-views

Hence I will keep it like this.

**Comment by bharitwal** [comment]
Minor edits from review comments have been completed.

**Review by irenkamalova — APPROVED** [review]
Amazing job! 👏

**Pull request merged**